Makefile

```diff
@@ -26,85 +26,94 @@
+IMAGE_PREFIX     ?= $(strip template-)
+IMAGE_SUFFIX     ?= $(strip )
 
 # Container registries.
-REGISTRY ?= cargo.dev.caicloud.xyz/release
+REGISTRY         ?= cargo.dev.caicloud.xyz/release
 
 # Container registry for base images.
-BASE_REGISTRY ?= cargo.caicloud.xyz/library
+BASE_REGISTRY    ?= cargo.caicloud.xyz/library
 
 #
 # These variables should not need tweaking.
 #
 
 # It's necessary to set this because some environments don't link sh -> bash.
-export SHELL := /bin/bash
+export SHELL     := /bin/bash
 
 # It's necessary to set the errexit flags for the bash shell.
 export SHELLOPTS := errexit
 
 # Project main package location (can be multiple ones).
-CMD_DIR := ./cmd
+CMD_DIR          := ./cmd
 
 # Project output directory.
-OUTPUT_DIR := ./bin
+OUTPUT_DIR       := ./bin
 
 # Build direcotory.
-BUILD_DIR := ./build
+BUILD_DIR        := ./build
 
 # Current version of the project.
-VERSION ?= $(shell git describe --tags --always --dirty)
+VERSION          ?= $(shell git describe --tags --always --dirty)
 
 # Available cpus for compiling, please refer to https://github.com/caicloud/engineering/issues/8186#issuecomment-518656946 for more information.
-CPUS ?= $(shell /bin/bash hack/read_cpus_available.sh)
+CPUS             ?= $(shell /bin/bash hack/read_cpus_available.sh)
 
 # Track code version with Docker Label.
-DOCKER_LABELS ?= git-describe="$(shell date -u +v%Y%m%d)-$(shell git describe --tags --always --dirty)"
+DOCKER_LABELS    ?= git-describe="$(shell date -u +v%Y%m%d)-$(shell git describe --tags --always --dirty)"
 
 # Golang standard bin directory.
-GOPATH ?= $(shell go env GOPATH)
-BIN_DIR := $(GOPATH)/bin
-GOLANGCI_LINT := $(BIN_DIR)/golangci-lint
+GOPATH           ?= $(shell go env GOPATH)
+BIN_DIR          := $(GOPATH)/bin
+GOLANGCI_LINT    := $(BIN_DIR)/golangci-lint
+
+ifeq ($(OS),Windows_NT)
+    OSName       ?= windows
+else
+    OSName       ?= $(shell echo $(shell uname -s) | tr '[:upper:]' '[:lower:]')
+endif
 
 #
 # Define all targets. At least the following commands are required:
 #
 
-# All targets.
-.PHONY: lint test build container push
-
+.PHONY: build
 build: build-local
 
 # more info about `GOGC` env: https://github.com/golangci/golangci-lint#memory-usage-of-golangci-lint
+.PHONY: lint
 lint: $(GOLANGCI_LINT)
 	@GOGC=5 $(GOLANGCI_LINT) run
 
+.PHONY: $(GOLANGCI_LINT)
 $(GOLANGCI_LINT):
 	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(BIN_DIR) v1.16.0
 
+.PHONY: test
 test:
 	@go test -p $(CPUS) $$(go list ./... | grep -v /vendor | grep -v /test) -coverprofile=coverage.out
 	@go tool cover -func coverage.out | tail -n 1 | awk '{ print "Total coverage: " $$3 }'
 
+.PHONY: build-local
 build-local:
 	@for target in $(TARGETS); do                                                      \
-	  go build -i -v -o $(OUTPUT_DIR)/$${target} -p $(CPUS)                            \
+	  env GOOS=$(OSName) go build -i -v -o $(OUTPUT_DIR)/$${target} -p $(CPUS)         \
```

Comment:
```
env OSName=linux make build-local
```
Now, I can build Linux binary on macOS.

Reply:
Every time build the linux binary in docker spend too long though i fix just one line.

`build-local` meaning build on my local env. Target `build-linux` is not useful.

Reply:
This change will make no difference, as `OSName` is the OS of current machine.

If you want to build for other OS, you can just use `env GOOS =linux make build-local`.

Reply:
Glad to see this.

I set all of variable and toolchains as Makefile envs in my others projects, so I can see them clearly.

Simple or explicit declarations, I prefer explicit, Rust tell us.